File: fabric/src/main/java/com/lunabbie/mushroommod/MushroomMod.java
package com.lunabbie.mushroommod;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class MushroomMod implements ModInitializer {

    public static final Item PAINSHROOM = new Item(new FabricItemSettings().group(ItemGroup.MISC));

    @Override
    public void onInitialize() {
        Registry.register(Registry.ITEM, new Identifier("mushroommod", "painshroom"), PAINSHROOM);
    }
}

File: fabric/src/main/java/com/lunabbie/mushroommod/MushroomMod.kt
package com.lunabbie.mushroommod

import com.lunabbie.mushroommod.items.groups.ModItemGroup
import net.minecraft.util.Identifier

const val MOD_ID = "mushroommod"
fun identifier(id: String) = Identifier.of(MOD_ID, id)

fun init() {
    ModItemGroup.registerAll()
    ModBlocks.register()
}

fun initClient() {
    ModBlocks.registerRendering()
}
